activesupport/lib/active_support/core_ext/enumerable.rb

```diff
@@ -261,16 +233,6 @@ def sum(identity = nil)
 }
 
 class Array #:nodoc:
-  # Array#sum was added in Ruby 2.4 but it only works with Numeric elements.
```

Comment:
@ermolaev I think this is the reason why rails still has an implementation of `Array#sum` 🤔

Reply:
I have not seen this used in the code, I think no one uses this feature

this case very specific
```ruby
class SummablePayment < Payment
  def +(p) self.class.new(price + p.price) end
end
```

if someone writes this, then it can also redefine sum via ruby refinements

Reply:
As someone who uses `sum` with objects I'm not happy with removing this feature but if we do, it definitely needs deprecating first as @eugeneius says.

Reply:
That is true, a very common use case are Money objects. They usually define `+` and it is common to want to `sum` them. I don't think we should be removing this feature.

Reply:
@rafaelfranca @pixeltrix So the idea is keep this feature or deprecate it? @ermolaev Have you an example of a use case where this method being 3x slowly is affecting the performance of your app? Asking because this performance improvement seems to me a bit premature since I cannot see a use case where this change will bring a considerable performance improvements to a rails app 🤔